bokehjs/src/lib/models/transforms/jitter.ts

```diff
@@ -35,10 +36,28 @@ export class Jitter extends RangeTransform {
   }
 
   override v_compute(xs0: Arrayable<number | Factor>): Arrayable<number> {
-    if (this.previous_values != null && this.previous_values.length == xs0.length)
-      return this.previous_values
-    this.previous_values = super.v_compute(xs0)
-    return this.previous_values
+    let xs: Arrayable<number>
+    if (super.range instanceof FactorRange)
+      xs = super.range.v_synthetic(xs0)
+    else if (isArrayableOf(xs0, isNumber))
+      xs = xs0
+    else
+      throw new Error("unexpected")
+
+    if (this.previous_offsets != null && this.previous_offsets.length == xs.length) {
+      const result = new (infer_type(xs))(xs.length)
+      for (let i = 0; i < xs.length; i++) {
+        result[i] = this.previous_offsets[i] + xs[i]
+      }
+      return result
```

Comment:
This whole block could be simplified with:
```ts
return map(xs, (xsi, i) => this.previous_offsets[i] + xsi)
```
from `core/util/arrayable`. `map()` preserves the length and type of the container. Similarly the code below could be simplified.

Reply:
Using `this` in arrow function didn't work, so I assigned `previous_offsets` to another variable to pass it into the arrow function.